In [1]:
import pandas as pd

# Listings Datasets

Loading listing datasets

In [2]:
list1 = pd.read_csv("CRMLSListing202602.csv")
list2 = pd.read_csv("CRMLSListing202603.csv")
list3 = pd.read_csv("CRMLSListing202604.csv")

C:\Users\mukun\AppData\Local\Temp\ipykernel_5988\1406942727.py:1: DtypeWarning: Columns (32,58,63,66) have mixed types. Specify dtype option on import or set low_memory=False.
  list1 = pd.read_csv("CRMLSListing202602.csv")


Checking size of each listing dataset and combined length before concatenation

In [3]:
len(list1)

32884

In [4]:
len(list2)

39153

In [5]:
len(list3)

39020

In [6]:
len(list1) + len(list2) + len(list3)

111057

Concatenating the listing datasets

In [7]:
listings = pd.concat([list1, list2, list3])

In [8]:
listings.head()

,OriginalListPrice,ListingKey,ListAgentEmail,CloseDate,ClosePrice,ListAgentFirstName,ListAgentLastName,Latitude,Longitude,UnparsedAddress,...,MainLevelBedrooms,NewConstructionYN,GarageSpaces,HighSchoolDistrict,PostalCode,BuyerOfficeName.1,AssociationFee,LotSizeSquareFeet,MiddleOrJuniorSchoolDistrict,UnparsedAddress.1
0,699000.0,1155102284,realtorck123@gmail.com,NaN,NaN,KWOK,CHOO,34.157413,-118.122500,1348 E Orange Grove Boulevard 4,...,1.0,False,2.0,Pasadena Unified,91104,NaN,456.0,13985.0,NaN,1348 E Orange Grove Boulevard 4
1,6395000.0,1155102282,tazzi@theazzigroup.com,NaN,NaN,Tony,Azzi,34.205933,-118.369947,7442 Vineland Avenue,...,NaN,False,NaN,NaN,91352,NaN,NaN,41013.0,NaN,7442 Vineland Avenue
2,2300000.0,1155098526,Marilyn@ocluxuryhomes.com,NaN,NaN,Marilyn,Ryder,33.476010,-117.701575,24444 Alta Vista Drive 19,...,1.0,False,2.0,Capistrano Unified,92629,NaN,891.0,0.0,NaN,24444 Alta Vista Drive 19
3,1100000.0,1155097469,ryan.bass@camoves.com,NaN,NaN,Ryan,Bass,32.807548,-117.020585,6849 Ballinger Avenue,...,3.0,NaN,2.0,San Diego Unified,92119,NaN,0.0,7200.0,NaN,6849 Ballinger Avenue
4,550000.0,1155097291,soldbycrystalhoggard@gmail.com,NaN,NaN,CRYSTAL,HOGGARD,34.247686,-117.321376,330 Canyon Drive,...,2.0,False,0.0,Rim of the World,92322,NaN,0.0,107526.0,NaN,330 Canyon Drive


Ensuring that the length of the concatentated dataset matches the length of the indivudal datasets combined

In [9]:
len(listings)

111057

In [10]:
listings.columns

Index(['OriginalListPrice', 'ListingKey', 'ListAgentEmail', 'CloseDate',
       'ClosePrice', 'ListAgentFirstName', 'ListAgentLastName', 'Latitude',
       'Longitude', 'UnparsedAddress', 'PropertyType', 'LivingArea',
       'ListPrice', 'DaysOnMarket', 'ListOfficeName', 'BuyerOfficeName',
       'CoListOfficeName', 'ListAgentFullName', 'CoListAgentFirstName',
       'CoListAgentLastName', 'BuyerAgentMlsId', 'BuyerAgentFirstName',
       'BuyerAgentLastName', 'FireplacesTotal', 'AssociationFeeFrequency',
       'AboveGradeFinishedArea', 'ListingKeyNumeric', 'MLSAreaMajor',
       'TaxAnnualAmount', 'CountyOrParish', 'PropertyType.1', 'MlsStatus',
       'ElementarySchool', 'ListAgentFirstName.1', 'AttachedGarageYN',
       'ParkingTotal', 'BuilderName', 'PropertySubType', 'LotSizeAcres',
       'SubdivisionName', 'BuyerOfficeAOR', 'YearBuilt', 'DaysOnMarket.1',
       'StreetNumberNumeric', 'LivingArea.1', 'ListingId',
       'BathroomsTotalInteger', 'City', 'TaxYear', 'BuildingAreaTot

Filtering for residential listings

In [11]:
listings_residential = listings[listings["PropertyType"] == "Residential"]

In [12]:
len(listings) - len(listings_residential)

38130

There were 38,130 non-residential properties in the sold dataset that were filtered out.

In [13]:
len(listings_residential)

72927

The combined length for listed residential properties is 72,927.

# Sold Datasets

Repeating same process for sold datasets

In [14]:
sold1 = pd.read_csv("CRMLSSold202602.csv")
sold2 = pd.read_csv("CRMLSSold202603.csv")
sold3 = pd.read_csv("CRMLSSold202604.csv")

In [15]:
len(sold1) + len(sold2) + len(sold3)

66643

In [16]:
sold = pd.concat([sold1, sold2, sold3])

In [17]:
len(sold)

66643

In [18]:
sold.columns

Index(['BuyerAgentAOR', 'ListAgentAOR', 'Flooring', 'ViewYN', 'WaterfrontYN',
       'BasementYN', 'PoolPrivateYN', 'OriginalListPrice', 'ListingKey',
       'ListAgentEmail', 'CloseDate', 'ClosePrice', 'ListAgentFirstName',
       'ListAgentLastName', 'Latitude', 'Longitude', 'UnparsedAddress',
       'PropertyType', 'LivingArea', 'ListPrice', 'DaysOnMarket',
       'ListOfficeName', 'BuyerOfficeName', 'CoListOfficeName',
       'ListAgentFullName', 'CoListAgentFirstName', 'CoListAgentLastName',
       'BuyerAgentMlsId', 'BuyerAgentFirstName', 'BuyerAgentLastName',
       'FireplacesTotal', 'AssociationFeeFrequency', 'AboveGradeFinishedArea',
       'ListingKeyNumeric', 'MLSAreaMajor', 'TaxAnnualAmount',
       'CountyOrParish', 'MlsStatus', 'ElementarySchool', 'AttachedGarageYN',
       'ParkingTotal', 'BuilderName', 'PropertySubType', 'LotSizeAcres',
       'SubdivisionName', 'BuyerOfficeAOR', 'YearBuilt', 'StreetNumberNumeric',
       'ListingId', 'BathroomsTotalInteger', 'City', '

Ensuring that the date ranges of the combined dataset matches expectations.

In [19]:
sold["CloseDate"].min()

'2026-02-01'

In [20]:
sold["CloseDate"].max()

'2026-04-30'

Filtering for only residential properties.

In [21]:
sold_residential = sold[sold["PropertyType"] == "Residential"]

In [22]:
len(sold) - len(sold_residential)

22134

There were 22,134 non-residential properties in the sold dataset that were filtered out.

In [23]:
len(sold_residential)

44509

The combined length for sold residential properties is 44,509.

## Saving Combined and Filtered Datasets as CSVs

In [24]:
listings_residential.to_csv("listings_residential.csv", index=False)

In [25]:
sold_residential.to_csv("sold_residential", index=False)